syntax = "proto3";
package queue;
option go_package = "github.com/abayleypublic/queue/gateway";

import "google/api/annotations.proto";

service Queue {
  rpc GetQueue (GetQueueRequest) returns (GetQueueResponse) {
    option (google.api.http) = {
      get: "/queue/{id}"
    };
  }

  rpc SetQueue (SetQueueRequest) returns (SetQueueResponse) {
    option (google.api.http) = {
      put: "/queue/{id}"
      body: "*"
    };
  }
}

message Entity {
    string id = 1;
    string name = 2;
}

message GetQueueRequest {
  string id = 1;
}

message GetQueueResponse {
  repeated Entity entities = 1;
}

message SetQueueRequest {
  string id = 1;
  repeated Entity entities = 2;
}

message SetQueueResponse {}